### PR TITLE
feat: enhance markdown preview for backlinks

### PR DESCRIPTION
**Pull request by pengx17**
fix AF-1770
fix AF-1771

---

fix: doc link middlewares
feat: markdown renderer
feat: allow multiple backlink for a single doc
feat: showing correct doc ref link
feat: trim long para & ident lists
feat: list indentition fix
feat: database/latext handling
feat: other block types handling
fix: lint

### CODE REVIEW

**Comment by pengx17** [comment]
* **#8957** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8957?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8956** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8956?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8956?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8883** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8883?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/modules/docs-search/schema.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ export type DocIndexSchema = typeof docIndexSchema;
 export const blockIndexSchema = defineSchema({
   docId: 'String',
   blockId: 'String',
+  // note id of the block. used for focusing the note in edgeless mode
+  noteBlockId: 'String',
```
Comment (EYHN) [inline packages/frontend/core/src/modules/docs-search/schema.ts]: can we put this to `additional` ?

Comment (pengx17) [reply packages/frontend/core/src/modules/docs-search/schema.ts]: sure

**Review by EYHN — APPROVED** [review]

File: packages/frontend/core/src/modules/docs-search/worker/in-worker.ts [hunk 2]
```diff
@@ -119,50 +89,369 @@ interface BlockDocumentInfo {
...
+        }
+      }
+    }
+
+    if (!databaseModel) {
+      return null;
+    }
+
+    const url = getDocLink(block.docId, databaseModel.id);
+    const title = databaseModel.title;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/docs-search/worker/in-worker.ts]: can we use `block.additional.databaseName` ?

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/bi-directional-link-panel.tsx [hunk 3]
```diff
@@ -1,32 +1,196 @@
...
+        'docLinkBaseUrl',
+        `/workspace/${workspaceService.workspace.id}`
+      );
+    };
+
+    // make sure the doc meta is available for the reference renderer
+    const rootDocMetaMiddleware: JobMiddleware = ({ collection, slots }) => {
+      slots.beforeImport.on(() => {
+        workspaceService.workspace.docCollection.docs.forEach(doc => {
+          if (doc.meta && !collection.meta.getDocMeta(doc.id)) {
```
Comment (EYHN) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/bi-directional-link-panel.tsx]: so we use AffinePageReference to render links, do we still need this?

**Pull request merged**